lntest/itest/lnd_res_handoff_test.go

```diff
@@ -0,0 +1,179 @@
+package itest
+
+import (
+	"time"
+
+	"github.com/btcsuite/btcutil"
+	"github.com/lightningnetwork/lnd/lnrpc"
+	"github.com/lightningnetwork/lnd/lntest"
+	"github.com/lightningnetwork/lnd/lntest/wait"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/net/context"
+)
+
+// testResHandoff tests that the contractcourt is able to properly hand-off
+// resolution messages to the switch.
+func testResHandoff(net *lntest.NetworkHarness, t *harnessTest) {
```

Comment:
Yeah I can add that, just need to hammer down the final design here

Reply:
Adding a test as initiator doesn't make sense as it wouldn't be possible to test the persistence logic in the same way as the multi-hop case where the resolution message was previously lost. There's no "hook" in the switch to prevent handing the resolution msg to the local sending / network result store code that could be used to test a restart case